freqtrade/freqtradebot.py

```diff
@@ -547,24 +548,32 @@ def leverage_prep(
         interest_rate = 0.0
         isolated_liq = None
 
-        # TODO-lev: Uncomment once liq and interest merged in
         # if TradingMode == TradingMode.MARGIN:
         #     interest_rate = self.exchange.get_interest_rate(
         #         pair=pair,
         #         open_rate=open_rate,
         #         is_short=is_short
         #     )
-
-        #     if self.collateral_type == Collateral.ISOLATED:
-
-        #         isolated_liq = liquidation_price(
-        #             exchange_name=self.exchange.name,
-        #             trading_mode=self.trading_mode,
-        #             open_rate=open_rate,
-        #             amount=amount,
-        #             leverage=leverage,
-        #             is_short=is_short
-        #         )
+        maintenance_amt, mm_rate = self.exchange.get_mm_amt_rate(pair, amount)
+
+        if self.collateral_type == Collateral.ISOLATED:
+            if self.config['dry_run']:
+                isolated_liq = liquidation_price(
+                    exchange_name=self.exchange.name,
+                    open_rate=open_rate,
+                    is_short=is_short,
+                    leverage=leverage,
+                    trading_mode=self.trading_mode,
+                    collateral=Collateral.ISOLATED,
+                    mm_ex_1=0.0,
+                    upnl_ex_1=0.0,
+                    position=amount * open_rate,
+                    wallet_balance=amount/leverage,  # TODO-lev: Is this correct?
+                    maintenance_amt=maintenance_amt,
+                    mm_rate=mm_rate,
+                )
```

Comment:
@frosty00 I'm not sure how to get the `maintenance_amt` and `mm_rate` properties, I wrote an exchange method `get_mm_amt_rate` but right now it does nothing. They're [explained here,](https://www.binance.com/en/support/faq/b3c689c1f50a44cabb3a84e663b81d93) and there's some tables for most of the binance ones.

I'm also not certain if my `wallet_balance` calculation is correct. Right now this should only work for isolated futures, so I figured that balance is whatever collateral you spend on the trade. Isolated futures (on Binance at least) don't require you to transfer money into the isolated account before you spend it, the collateral get's transferred into it when you make a purchase.

Reply:
@frosty00 Any update here?

Reply:
The fields on okex and gateio which I'm having trouble finding are `mm_ratio`, `liability`, `interest`, `position_assets`

Reply:
Alrighty ✊